### PR TITLE
fix(error): shows error in output channel

### PR DESCRIPTION
**Pull request by invincibleJai**
Tracks: https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues/281

### CODE REVIEW

File: src/DepOutputChannel.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+import * as vscode from 'vscode';
+
+export class DepOutputChannel {
+  outputChannel: vscode.OutputChannel;
+  constructor(channelName = 'Dependency Analytics') {
```
Comment (sudhirverma) [inline src/DepOutputChannel.ts]: add functionality to get the `Dependency Analytics` command from the command palette.

File: src/DepOutputChannel.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  getOutputChannel(): vscode.OutputChannel {
+    this.outputChannel.clear();
+    return this.outputChannel;
+  }
+
+  showOutputChannel() {
+    this.outputChannel.show();
+  }
+
+  clearOutputChannel() {
```
Comment (sudhirverma) [inline src/DepOutputChannel.ts]: why are you clearing the output channel?

File: src/DepOutputChannel.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      this.outputChannel = vscode.window.createOutputChannel(channelName);
+    }
+  }
+
+  getOutputChannel(): vscode.OutputChannel {
+    this.outputChannel.clear();
+    return this.outputChannel;
+  }
+
+  showOutputChannel() {
```
Comment (sudhirverma) [inline src/DepOutputChannel.ts]: and provide type for the function

File: src/DepOutputChannel.ts (hunk 2 above)
Comment (invincibleJai) [reply src/DepOutputChannel.ts]: Am making use of outputChannel to show `stdout/stderr` for each stack analysis so clearing would make sense before each run

File: src/DepOutputChannel.ts (hunk 1 above)
Comment (invincibleJai) [reply src/DepOutputChannel.ts]: i didn't get you, it's just the Output channel default name, what it has to with Command palette.

File: src/DepOutputChannel.ts (hunk 3 above)
Comment (invincibleJai) [reply src/DepOutputChannel.ts]: You mean return type i.e `void` in this scenario?

File: src/DepOutputChannel.ts (hunk 2 above)
Comment (sudhirverma) [reply src/DepOutputChannel.ts]: you can write eg: https://github.com/redhat-developer/vscode-adapters/blob/master/src/extension.ts#L17 for stdout/stderr

File: src/DepOutputChannel.ts (hunk 3 above)
Comment (sudhirverma) [reply src/DepOutputChannel.ts]: yes

File: src/DepOutputChannel.ts (hunk 2 above)
Comment (invincibleJai) [reply src/DepOutputChannel.ts]: Currently am writing stdout/stderr in same channel as in https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/pull/284/files#diff-55eabb84e751500cb765a140d212c608R191 , i don't think we need two channel's here

File: src/DepOutputChannel.ts (hunk 1 above)
Comment (sudhirverma) [reply src/DepOutputChannel.ts]: what if I want to see the `Dependency Analytics` log from command Palette?

**Comment by invincibleJai** [comment]
[test]

**Comment by invincibleJai** [comment]
[test]

File: src/DepOutputChannel.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  showOutputChannel(): void {
+    this.outputChannel.show();
+  }
+
+  clearOutputChannel(): void {
+    this.outputChannel.clear();
+  }
+
+  addMsgOutputChannel(msg: string) {
```
Comment (sudhirverma) [inline src/DepOutputChannel.ts]: type is missing for this one

**Comment by invincibleJai** [comment]
[test]

**Comment by invincibleJai** [comment]
[test]

**Comment by invincibleJai** [comment]
[test]

File: src/extension.ts [hunk 5]
```diff
@@ -95,27 +94,39 @@ export function activate(context: vscode.ExtensionContext) {
...
+                  vscode.window.activeTextEditor &&
+                  ((respData.diagCount > 0 &&
+                    respData.diagCount !== diagCountInfo) ||
+                    !onFileOpen ||
+                    (onFileOpen &&
+                      onFileOpen.indexOf(
+                        vscode.window.activeTextEditor.document.fileName
+                      ) === -1))
+                ) {
+                  setTimeout(() => {
```
Comment (sudhirverma) [inline src/extension.ts]: why are you using setTimeout?

Comment (invincibleJai) [reply src/extension.ts]: `setTimeout` is used here to resolve the status after 1.5 secs as in this case notification are coming via LS based on various states

**Pull request merged**